Mac OS upscale beyond 2k issue

i can upscale to 2k but no further on an M series mchine with 96GB unified Ram
Surely with more way more than the 16GB vRam must cuda cards have or even the max of 32GB of a 5090 a M series Mac should be able to easily upscale to 8k or even higher?


# ComfyUI Error Report
## Error Details
- **Node ID:** 10
- **Node Type:** SeedVR2VideoUpscaler
- **Exception Type:** RuntimeError
- **Exception Message:** MPS backend out of memory (MPS allocated: 18.69 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).

## Stack Trace
```
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 515, in execute
    output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 329, in get_output_data
    return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 303, in _async_map_node_over_list
    await process_inputs(input_dict, i)

  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 291, in process_inputs
    result = f(**inputs)
             ^^^^^^^^^^^

  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/comfy_api/internal/__init__.py", line 149, in wrapped_func
    return method(locked_class, **inputs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/comfy_api/latest/_io.py", line 1513, in EXECUTE_NORMALIZED
    to_return = cls.execute(*args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/interfaces/video_upscaler.py", line 580, in execute
    raise e

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/interfaces/video_upscaler.py", line 497, in execute
    ctx = decode_all_batches(
          ^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/core/generation_phases.py", line 937, in decode_all_batches
    samples = runner.vae_decode([upscaled_latent])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/core/infer.py", line 262, in vae_decode
    sample = self.vae.decode(
             ^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1688, in decode
    x = super().decode(z, return_dict=return_dict, tiled=tiled, tile_size=tile_size,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/diffusers/utils/accelerate_utils.py", line 46, in wrapper
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1203, in decode
    decoded = self.tiled_decode(z, tile_size=tile_size, tile_overlap=tile_overlap)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1557, in tiled_decode
    decoded_tile = self.slicing_decode(tile_latent)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1292, in slicing_decode
    return self._decode(z)
           ^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1240, in _decode
    output = self.decoder(_z, memory_state=memory_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1028, in forward
    sample = up_block(sample, latent_embeds, memory_state=memory_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 553, in forward
    hidden_states = resnet(hidden_states, temb=None, memory_state=memory_state)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 336, in forward
    hidden_states = self.conv1(hidden_states, memory_state=memory_state)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 227, in forward
    return self.slicing_forward(input, memory_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 296, in slicing_forward
    input[i] = self.memory_limit_conv(
               ^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 204, in memory_limit_conv
    output = retry_on_oom(
             ^^^^^^^^^^^^^

  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/optimization/memory_manager.py", line 394, in retry_on_oom
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^

```
## System Information
- **ComfyUI Version:** 0.4.0
- **Arguments:** /Applications/ComfyUI.app/Contents/Resources/ComfyUI/main.py --user-directory /Users/peter/Documents/ComfyUI/user --input-directory /Users/peter/Documents/ComfyUI/input --output-directory /Users/peter/Documents/ComfyUI/output --front-end-root /Applications/ComfyUI.app/Contents/Resources/ComfyUI/web_custom_versions/desktop_app --base-directory /Users/peter/Documents/ComfyUI --extra-model-paths-config /Users/peter/Library/Application Support/ComfyUI/extra_models_config.yaml --log-stdout --listen 127.0.0.1 --port 8000 --enable-manager
- **OS:** darwin
- **Python Version:** 3.12.11 (main, Aug 18 2025, 19:02:39) [Clang 20.1.4 ]
- **Embedded Python:** false
- **PyTorch Version:** 2.5.1
## Devices

- **Name:** mps
  - **Type:** mps
  - **VRAM Total:** 103079215104
  - **VRAM Free:** 70226493440
  - **Torch VRAM Total:** 103079215104
  - **Torch VRAM Free:** 70226493440

## Logs
```
2025-12-13T19:37:19.931399 -    ███████╗███████╗███████╗██████╗ ██╗   ██╗██████╗     ██████╗       ███████╗2025-12-13T19:37:19.931411 - 
2025-12-13T19:37:19.931572 -    ██╔════╝██╔════╝██╔════╝██╔══██╗██║   ██║██╔══██╗    ╚════██╗      ██╔════╝2025-12-13T19:37:19.931589 - 
2025-12-13T19:37:19.931607 -    ███████╗█████╗  █████╗  ██║  ██║██║   ██║██████╔╝     █████╔╝      ███████╗2025-12-13T19:37:19.931621 - 
2025-12-13T19:37:19.931636 -    ╚════██║██╔══╝  ██╔══╝  ██║  ██║╚██╗ ██╔╝██╔══██╗    ██╔═══╝       ╚════██║2025-12-13T19:37:19.931647 - 
2025-12-13T19:37:19.931658 -    ███████║███████╗███████╗██████╔╝ ╚████╔╝ ██║  ██║    ███████╗  ██╗ ███████║2025-12-13T19:37:19.931667 - 
2025-12-13T19:37:19.931677 -    ╚══════╝╚══════╝╚══════╝╚═════╝   ╚═══╝  ╚═╝  ╚═╝    ╚══════╝  ╚═╝ ╚══════╝2025-12-13T19:37:19.931685 - 
2025-12-13T19:37:19.931697 -    v2.5.22                                    © ByteDance Seed · NumZ · AInVFX2025-12-13T19:37:19.931706 - 
2025-12-13T19:37:19.931717 -    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━2025-12-13T19:37:19.931726 - 
2025-12-13T19:37:19.931736 -  2025-12-13T19:37:19.931744 - 
2025-12-13T19:37:19.932014 - [19:37:19.931] ⚠️ [WARNING] BlockSwap disabled: macOS uses unified memory (no separate VRAM/RAM). Ignoring blocks_to_swap=36, swap_io_components=False2025-12-13T19:37:19.932025 - 
2025-12-13T19:37:19.932048 - [19:37:19.932] 🏃 Creating new runner: DiT=seedvr2_ema_7b_sharp_fp16.safetensors, VAE=ema_vae_fp16.safetensors2025-12-13T19:37:19.932057 - 
2025-12-13T19:37:19.940893 - [19:37:19.940] 🚀 Creating DiT model structure on meta device2025-12-13T19:37:19.940913 - 
2025-12-13T19:37:20.022555 - [19:37:20.022] 🎨 Creating VAE model structure on meta device2025-12-13T19:37:20.022589 - 
2025-12-13T19:37:20.070326 - [19:37:20.070]  2025-12-13T19:37:20.070420 - 
2025-12-13T19:37:20.070467 - [19:37:20.070] 🎬 Starting upscaling generation...2025-12-13T19:37:20.070483 - 
2025-12-13T19:37:20.070523 - [19:37:20.070] 🎬   Input: 1 frame, 515x281px → Padded: 1024x560px → Output: 1024x558px (shortest edge: 1024px, max edge: 1024px)2025-12-13T19:37:20.070535 - 
2025-12-13T19:37:20.070551 - [19:37:20.070] 🎬   Batch size: 1, Seed: 42, Channels: RGBA2025-12-13T19:37:20.070561 - 
2025-12-13T19:37:20.070579 - [19:37:20.070]  2025-12-13T19:37:20.070588 - 
2025-12-13T19:37:20.070602 - [19:37:20.070]  ━━━━━━━━ Phase 1: VAE encoding ━━━━━━━━2025-12-13T19:37:20.070611 - 
2025-12-13T19:37:20.071967 - [19:37:20.071] 🎨 Materializing VAE weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/ema_vae_fp16.safetensors2025-12-13T19:37:20.071999 - 
2025-12-13T19:37:20.268955 - [19:37:20.268] 🎨 Encoding batch 1/12025-12-13T19:37:20.268986 - 
2025-12-13T19:37:20.269808 - [19:37:20.269] 📹   Sequence of 1 frames2025-12-13T19:37:20.269821 - 
2025-12-13T19:37:21.111187 - FETCH ComfyRegistry Data: 65/112
2025-12-13T19:37:24.504160 - FETCH ComfyRegistry Data: 70/112
2025-12-13T19:37:25.471750 - [19:37:25.471]  2025-12-13T19:37:25.471811 - 
2025-12-13T19:37:25.471867 - [19:37:25.471]  ━━━━━━━━ Phase 2: DiT upscaling ━━━━━━━━2025-12-13T19:37:25.471961 - 
2025-12-13T19:37:25.482223 - [19:37:25.482] 🚀 Materializing DiT weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/seedvr2_ema_7b_sharp_fp16.safetensors2025-12-13T19:37:25.482277 - 
2025-12-13T19:37:25.551903 - [19:37:25.551] 🔧 Converting NaDiT parameters/buffers to torch.bfloat16 for MPS backend2025-12-13T19:37:25.551936 - 
2025-12-13T19:37:28.042707 - FETCH ComfyRegistry Data: 75/112
2025-12-13T19:37:31.481020 - FETCH ComfyRegistry Data: 80/112
2025-12-13T19:37:35.065087 - FETCH ComfyRegistry Data: 85/112
2025-12-13T19:37:38.527469 - [19:37:38.527] 🎬 Upscaling batch 1/12025-12-13T19:37:38.527621 - 
2025-12-13T19:37:38.529415 - FETCH ComfyRegistry Data: 90/112
2025-12-13T19:37:38.739208 - 
EulerSampler:   0%|          | 0/1 [00:00<?, ?it/s]2025-12-13T19:37:41.986920 - FETCH ComfyRegistry Data: 95/112
2025-12-13T19:37:45.229572 - 
EulerSampler: 100%|██████████| 1/1 [00:06<00:00,  6.49s/it]2025-12-13T19:37:45.229886 - 
EulerSampler: 100%|██████████| 1/1 [00:06<00:00,  6.49s/it]2025-12-13T19:37:45.229923 - 
2025-12-13T19:37:45.237781 - [19:37:45.237]  2025-12-13T19:37:45.237808 - 
2025-12-13T19:37:45.237830 - [19:37:45.237]  ━━━━━━━━ Phase 3: VAE decoding ━━━━━━━━2025-12-13T19:37:45.237841 - 
2025-12-13T19:37:45.237886 - [19:37:45.237] 🔧 Pre-allocating output tensor: 1 frames, 1024x558px, RGBA (0.00GB)2025-12-13T19:37:45.237896 - 
2025-12-13T19:37:45.366367 - [19:37:45.366] 🎨 Decoding batch 1/12025-12-13T19:37:45.366445 - 
2025-12-13T19:37:45.499205 - FETCH ComfyRegistry Data: 100/112
2025-12-13T19:37:49.015073 - FETCH ComfyRegistry Data: 105/112
2025-12-13T19:37:52.485623 - FETCH ComfyRegistry Data: 110/112
2025-12-13T19:37:54.372697 - FETCH ComfyRegistry Data [DONE]
2025-12-13T19:37:54.459773 - [ComfyUI-Manager] default cache updated: https://api.comfy.org/nodes
2025-12-13T19:37:54.465165 - FETCH DATA from: /Users/peter/Documents/ComfyUI/user/__manager/cache/1514988643_custom-node-list.json2025-12-13T19:37:54.465182 - 2025-12-13T19:37:54.469753 -  [DONE]2025-12-13T19:37:54.469772 - 
2025-12-13T19:37:54.481982 - [ComfyUI-Manager] All startup tasks have been completed.
2025-12-13T19:37:57.295275 - [19:37:57.294]  2025-12-13T19:37:57.295326 - 
2025-12-13T19:37:57.295357 - [19:37:57.295]  ━━━━━━━━ Phase 4: Post-processing ━━━━━━━━2025-12-13T19:37:57.295373 - 
2025-12-13T19:37:57.295405 - [19:37:57.295] 👻 Processing Alpha batch 1/12025-12-13T19:37:57.295417 - 
2025-12-13T19:37:59.883572 - [19:37:59.883] 📹 Post-processing batch 1/12025-12-13T19:37:59.883722 - 
2025-12-13T19:38:00.043425 - [19:38:00.043] 📹   Applying LAB perceptual color transfer2025-12-13T19:38:00.043466 - 
2025-12-13T19:38:02.563729 - [19:38:02.563] 🎬 Output assembled: 1 frames, Resolution: 1024x558px, Channels: RGBA2025-12-13T19:38:02.563797 - 
2025-12-13T19:38:02.564113 - [19:38:02.563]  2025-12-13T19:38:02.564256 - 
2025-12-13T19:38:02.566636 - [19:38:02.566] ✅ Upscaling completed successfully!2025-12-13T19:38:02.566653 - 
2025-12-13T19:38:03.024373 - [19:38:03.024] ⚡ Average FPS: 0.02 frames/sec2025-12-13T19:38:03.024419 - 
2025-12-13T19:38:03.024456 - [19:38:03.024]  2025-12-13T19:38:03.024470 - 
2025-12-13T19:38:03.024491 - [19:38:03.024]  ────────────────────────2025-12-13T19:38:03.024503 - 
2025-12-13T19:38:03.024522 - [19:38:03.024] 💬 Questions? Updates? Watch the videos, star the repo & join us!2025-12-13T19:38:03.024534 - 
2025-12-13T19:38:03.024552 - [19:38:03.024] 🎬 https://www.youtube.com/@AInVFX2025-12-13T19:38:03.024563 - 
2025-12-13T19:38:03.024579 - [19:38:03.024] ⭐ https://github.com/numz/ComfyUI-SeedVR2_VideoUpscaler2025-12-13T19:38:03.024590 - 
2025-12-13T19:38:03.186307 - Prompt executed in 43.26 seconds
2025-12-13T19:38:35.950545 - got prompt
2025-12-13T19:38:35.955804 -  2025-12-13T19:38:35.955871 - 
2025-12-13T19:38:35.955926 -  2025-12-13T19:38:35.955965 - 
2025-12-13T19:38:35.956039 -    ███████╗███████╗███████╗██████╗ ██╗   ██╗██████╗     ██████╗       ███████╗2025-12-13T19:38:35.956055 - 
2025-12-13T19:38:35.956072 -    ██╔════╝██╔════╝██╔════╝██╔══██╗██║   ██║██╔══██╗    ╚════██╗      ██╔════╝2025-12-13T19:38:35.956127 - 
2025-12-13T19:38:35.956155 -    ███████╗█████╗  █████╗  ██║  ██║██║   ██║██████╔╝     █████╔╝      ███████╗2025-12-13T19:38:35.956169 - 
2025-12-13T19:38:35.956183 -    ╚════██║██╔══╝  ██╔══╝  ██║  ██║╚██╗ ██╔╝██╔══██╗    ██╔═══╝       ╚════██║2025-12-13T19:38:35.956195 - 
2025-12-13T19:38:35.956207 -    ███████║███████╗███████╗██████╔╝ ╚████╔╝ ██║  ██║    ███████╗  ██╗ ███████║2025-12-13T19:38:35.956216 - 
2025-12-13T19:38:35.956227 -    ╚══════╝╚══════╝╚══════╝╚═════╝   ╚═══╝  ╚═╝  ╚═╝    ╚══════╝  ╚═╝ ╚══════╝2025-12-13T19:38:35.956236 - 
2025-12-13T19:38:35.956249 -    v2.5.22                                    © ByteDance Seed · NumZ · AInVFX2025-12-13T19:38:35.956257 - 
2025-12-13T19:38:35.956268 -    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━2025-12-13T19:38:35.956276 - 
2025-12-13T19:38:35.956286 -  2025-12-13T19:38:35.956294 - 
2025-12-13T19:38:35.956838 - [19:38:35.956] ⚠️ [WARNING] BlockSwap disabled: macOS uses unified memory (no separate VRAM/RAM). Ignoring blocks_to_swap=36, swap_io_components=False2025-12-13T19:38:35.956850 - 
2025-12-13T19:38:35.956880 - [19:38:35.956] 🏃 Creating new runner: DiT=seedvr2_ema_7b_sharp_fp16.safetensors, VAE=ema_vae_fp16.safetensors2025-12-13T19:38:35.956890 - 
2025-12-13T19:38:35.965737 - [19:38:35.965] 🚀 Creating DiT model structure on meta device2025-12-13T19:38:35.965761 - 
2025-12-13T19:38:36.076598 - [19:38:36.076] 🎨 Creating VAE model structure on meta device2025-12-13T19:38:36.076635 - 
2025-12-13T19:38:36.172733 - [19:38:36.172]  2025-12-13T19:38:36.172779 - 
2025-12-13T19:38:36.172805 - [19:38:36.172] 🎬 Starting upscaling generation...2025-12-13T19:38:36.172816 - 
2025-12-13T19:38:36.172841 - [19:38:36.172] 🎬   Input: 1 frame, 515x281px → Padded: 2048x1120px → Output: 2048x1118px (shortest edge: 2048px, max edge: 2048px)2025-12-13T19:38:36.172852 - 
2025-12-13T19:38:36.172867 - [19:38:36.172] 🎬   Batch size: 1, Seed: 42, Channels: RGBA2025-12-13T19:38:36.172876 - 
2025-12-13T19:38:36.172898 - [19:38:36.172]  2025-12-13T19:38:36.172906 - 
2025-12-13T19:38:36.172920 - [19:38:36.172]  ━━━━━━━━ Phase 1: VAE encoding ━━━━━━━━2025-12-13T19:38:36.172929 - 
2025-12-13T19:38:36.174465 - [19:38:36.174] 🎨 Materializing VAE weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/ema_vae_fp16.safetensors2025-12-13T19:38:36.174487 - 
2025-12-13T19:38:36.603080 - [19:38:36.603] 🎨 Encoding batch 1/12025-12-13T19:38:36.603112 - 
2025-12-13T19:38:36.604116 - [19:38:36.604] 📹   Sequence of 1 frames2025-12-13T19:38:36.604149 - 
2025-12-13T19:38:36.671537 - [19:38:36.671] 🎨   Using VAE tiled encoding (Tile: (1024, 1024), Overlap: (128, 128))2025-12-13T19:38:36.671572 - 
2025-12-13T19:39:03.165054 - [19:39:03.164]  2025-12-13T19:39:03.165089 - 
2025-12-13T19:39:03.165112 - [19:39:03.165]  ━━━━━━━━ Phase 2: DiT upscaling ━━━━━━━━2025-12-13T19:39:03.165124 - 
2025-12-13T19:39:03.166031 - [19:39:03.165] 🚀 Materializing DiT weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/seedvr2_ema_7b_sharp_fp16.safetensors2025-12-13T19:39:03.166060 - 
2025-12-13T19:39:03.290145 - [19:39:03.290] 🔧 Converting NaDiT parameters/buffers to torch.bfloat16 for MPS backend2025-12-13T19:39:03.290185 - 
2025-12-13T19:39:07.410456 - [19:39:07.410] 🎬 Upscaling batch 1/12025-12-13T19:39:07.410541 - 
2025-12-13T19:39:27.573902 - 
EulerSampler: 100%|██████████| 1/1 [00:20<00:00, 20.08s/it]2025-12-13T19:39:27.574021 - 
EulerSampler: 100%|██████████| 1/1 [00:20<00:00, 20.08s/it]2025-12-13T19:39:27.574039 - 
2025-12-13T19:39:27.581542 - [19:39:27.581]  2025-12-13T19:39:27.581568 - 
2025-12-13T19:39:27.581597 - [19:39:27.581]  ━━━━━━━━ Phase 3: VAE decoding ━━━━━━━━2025-12-13T19:39:27.581609 - 
2025-12-13T19:39:27.581636 - [19:39:27.581] 🔧 Pre-allocating output tensor: 1 frames, 2048x1118px, RGBA (0.02GB)2025-12-13T19:39:27.581647 - 
2025-12-13T19:39:27.698511 - [19:39:27.698] 🎨 Decoding batch 1/12025-12-13T19:39:27.698551 - 
2025-12-13T19:39:27.707760 - [19:39:27.707] 🎨   Using VAE tiled decoding (Tile: (1024, 1024), Overlap: (128, 128))2025-12-13T19:39:27.707790 - 
2025-12-13T19:40:35.098057 - [19:40:35.097]  2025-12-13T19:40:35.098108 - 
2025-12-13T19:40:35.098137 - [19:40:35.098]  ━━━━━━━━ Phase 4: Post-processing ━━━━━━━━2025-12-13T19:40:35.098151 - 
2025-12-13T19:40:35.098176 - [19:40:35.098] 👻 Processing Alpha batch 1/12025-12-13T19:40:35.098187 - 
2025-12-13T19:40:36.324010 - [19:40:36.323] 📹 Post-processing batch 1/12025-12-13T19:40:36.324174 - 
2025-12-13T19:40:36.370092 - [19:40:36.370] 📹   Applying LAB perceptual color transfer2025-12-13T19:40:36.370129 - 
2025-12-13T19:40:37.313070 - [19:40:37.312] 🎬 Output assembled: 1 frames, Resolution: 2048x1118px, Channels: RGBA2025-12-13T19:40:37.313266 - 
2025-12-13T19:40:37.313438 - [19:40:37.313]  2025-12-13T19:40:37.313458 - 
2025-12-13T19:40:37.357877 - [19:40:37.357] ✅ Upscaling completed successfully!2025-12-13T19:40:37.357919 - 
2025-12-13T19:40:38.119201 - [19:40:38.119] ⚡ Average FPS: 0.01 frames/sec2025-12-13T19:40:38.119241 - 
2025-12-13T19:40:38.119328 - [19:40:38.119]  2025-12-13T19:40:38.119358 - 
2025-12-13T19:40:38.119386 - [19:40:38.119]  ────────────────────────2025-12-13T19:40:38.119398 - 
2025-12-13T19:40:38.119415 - [19:40:38.119] 💬 Questions? Updates? Watch the videos, star the repo & join us!2025-12-13T19:40:38.119426 - 
2025-12-13T19:40:38.119440 - [19:40:38.119] 🎬 https://www.youtube.com/@AInVFX2025-12-13T19:40:38.119450 - 
2025-12-13T19:40:38.119464 - [19:40:38.119] ⭐ https://github.com/numz/ComfyUI-SeedVR2_VideoUpscaler2025-12-13T19:40:38.119472 - 
2025-12-13T19:40:39.014359 - Prompt executed in 123.06 seconds
2025-12-13T19:42:48.705619 - got prompt
2025-12-13T19:42:48.710753 -  2025-12-13T19:42:48.711015 - 
2025-12-13T19:42:48.711064 -  2025-12-13T19:42:48.711098 - 
2025-12-13T19:42:48.711200 -    ███████╗███████╗███████╗██████╗ ██╗   ██╗██████╗     ██████╗       ███████╗2025-12-13T19:42:48.711294 - 
2025-12-13T19:42:48.711329 -    ██╔════╝██╔════╝██╔════╝██╔══██╗██║   ██║██╔══██╗    ╚════██╗      ██╔════╝2025-12-13T19:42:48.711344 - 
2025-12-13T19:42:48.711359 -    ███████╗█████╗  █████╗  ██║  ██║██║   ██║██████╔╝     █████╔╝      ███████╗2025-12-13T19:42:48.711371 - 
2025-12-13T19:42:48.711384 -    ╚════██║██╔══╝  ██╔══╝  ██║  ██║╚██╗ ██╔╝██╔══██╗    ██╔═══╝       ╚════██║2025-12-13T19:42:48.711394 - 
2025-12-13T19:42:48.711406 -    ███████║███████╗███████╗██████╔╝ ╚████╔╝ ██║  ██║    ███████╗  ██╗ ███████║2025-12-13T19:42:48.711416 - 
2025-12-13T19:42:48.711427 -    ╚══════╝╚══════╝╚══════╝╚═════╝   ╚═══╝  ╚═╝  ╚═╝    ╚══════╝  ╚═╝ ╚══════╝2025-12-13T19:42:48.711444 - 
2025-12-13T19:42:48.711458 -    v2.5.22                                    © ByteDance Seed · NumZ · AInVFX2025-12-13T19:42:48.711469 - 
2025-12-13T19:42:48.711483 -    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━2025-12-13T19:42:48.711493 - 
2025-12-13T19:42:48.711504 -  2025-12-13T19:42:48.711514 - 
2025-12-13T19:42:48.712397 - [19:42:48.712] ⚠️ [WARNING] BlockSwap disabled: macOS uses unified memory (no separate VRAM/RAM). Ignoring blocks_to_swap=36, swap_io_components=False2025-12-13T19:42:48.712420 - 
2025-12-13T19:42:48.712449 - [19:42:48.712] 🏃 Creating new runner: DiT=seedvr2_ema_7b_sharp_fp16.safetensors, VAE=ema_vae_fp16.safetensors2025-12-13T19:42:48.712501 - 
2025-12-13T19:42:48.721773 - [19:42:48.721] 🚀 Creating DiT model structure on meta device2025-12-13T19:42:48.721802 - 
2025-12-13T19:42:48.805206 - [19:42:48.805] 🎨 Creating VAE model structure on meta device2025-12-13T19:42:48.805244 - 
2025-12-13T19:42:49.040948 - [19:42:49.040]  2025-12-13T19:42:49.040989 - 
2025-12-13T19:42:49.041024 - [19:42:49.041] 🎬 Starting upscaling generation...2025-12-13T19:42:49.041036 - 
2025-12-13T19:42:49.041062 - [19:42:49.041] 🎬   Input: 1 frame, 515x281px → Padded: 4096x2240px → Output: 4096x2234px (shortest edge: 4096px, max edge: 4096px)2025-12-13T19:42:49.041075 - 
2025-12-13T19:42:49.041091 - [19:42:49.041] 🎬   Batch size: 1, Seed: 42, Channels: RGBA2025-12-13T19:42:49.041102 - 
2025-12-13T19:42:49.041119 - [19:42:49.041]  2025-12-13T19:42:49.041128 - 
2025-12-13T19:42:49.041159 - [19:42:49.041]  ━━━━━━━━ Phase 1: VAE encoding ━━━━━━━━2025-12-13T19:42:49.041169 - 
2025-12-13T19:42:49.042550 - [19:42:49.042] 🎨 Materializing VAE weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/ema_vae_fp16.safetensors2025-12-13T19:42:49.042570 - 
2025-12-13T19:42:49.802419 - [19:42:49.802] 🎨 Encoding batch 1/12025-12-13T19:42:49.807268 - 
2025-12-13T19:42:49.808236 - [19:42:49.808] 📹   Sequence of 1 frames2025-12-13T19:42:49.808260 - 
2025-12-13T19:42:49.877060 - [19:42:49.876] 🎨   Using VAE tiled encoding (Tile: (1024, 1024), Overlap: (128, 128))2025-12-13T19:42:49.877104 - 
2025-12-13T19:44:19.000639 - [19:44:19.000]  2025-12-13T19:44:19.000719 - 
2025-12-13T19:44:19.000763 - [19:44:19.000]  ━━━━━━━━ Phase 2: DiT upscaling ━━━━━━━━2025-12-13T19:44:19.000778 - 
2025-12-13T19:44:19.002195 - [19:44:19.002] 🚀 Materializing DiT weights to CPU (offload device): /Users/peter/Documents/ComfyUI/models/SEEDVR2/seedvr2_ema_7b_sharp_fp16.safetensors2025-12-13T19:44:19.002211 - 
2025-12-13T19:44:19.075249 - [19:44:19.075] 🔧 Converting NaDiT parameters/buffers to torch.bfloat16 for MPS backend2025-12-13T19:44:19.075277 - 
2025-12-13T19:44:30.528702 - [19:44:30.528] 🎬 Upscaling batch 1/12025-12-13T19:44:30.528770 - 
2025-12-13T19:45:40.596600 - 
EulerSampler: 100%|██████████| 1/1 [01:09<00:00, 69.99s/it]2025-12-13T19:45:40.596863 - 
EulerSampler: 100%|██████████| 1/1 [01:09<00:00, 69.99s/it]2025-12-13T19:45:40.596896 - 
2025-12-13T19:45:40.615241 - [19:45:40.615]  2025-12-13T19:45:40.615271 - 
2025-12-13T19:45:40.615292 - [19:45:40.615]  ━━━━━━━━ Phase 3: VAE decoding ━━━━━━━━2025-12-13T19:45:40.615305 - 
2025-12-13T19:45:40.615331 - [19:45:40.615] 🔧 Pre-allocating output tensor: 1 frames, 4096x2234px, RGBA (0.07GB)2025-12-13T19:45:40.615343 - 
2025-12-13T19:45:40.763170 - [19:45:40.763] 🎨 Decoding batch 1/12025-12-13T19:45:40.763210 - 
2025-12-13T19:45:40.775063 - [19:45:40.775] 🎨   Using VAE tiled decoding (Tile: (1024, 1024), Overlap: (128, 128))2025-12-13T19:45:40.775098 - 
2025-12-13T19:47:27.575529 - [19:47:27.574] ⚠️ [WARNING] OOM during InflatedCausalConv3d.concat_splits: MPS backend out of memory (MPS allocated: 18.70 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).2025-12-13T19:47:27.575589 - 
2025-12-13T19:47:27.575650 - [19:47:27.575] ℹ️ Clearing memory and retrying2025-12-13T19:47:27.575675 - 
2025-12-13T19:47:28.245612 - [19:47:28.245] ❌ [ERROR] Retry failed for InflatedCausalConv3d.concat_splits: MPS backend out of memory (MPS allocated: 18.69 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).2025-12-13T19:47:28.245784 - 
2025-12-13T19:47:28.246026 - [19:47:28.245] ❌ [ERROR] Error in Phase 3 (Decoding): MPS backend out of memory (MPS allocated: 18.69 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).2025-12-13T19:47:28.246106 - 
2025-12-13T19:47:28.394382 - !!! Exception during processing !!! MPS backend out of memory (MPS allocated: 18.69 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).
2025-12-13T19:47:28.402280 - Traceback (most recent call last):
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/optimization/memory_manager.py", line 376, in retry_on_oom
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
RuntimeError: MPS backend out of memory (MPS allocated: 18.70 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 515, in execute
    output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 329, in get_output_data
    return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 303, in _async_map_node_over_list
    await process_inputs(input_dict, i)
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/execution.py", line 291, in process_inputs
    result = f(**inputs)
             ^^^^^^^^^^^
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/comfy_api/internal/__init__.py", line 149, in wrapped_func
    return method(locked_class, **inputs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Applications/ComfyUI.app/Contents/Resources/ComfyUI/comfy_api/latest/_io.py", line 1513, in EXECUTE_NORMALIZED
    to_return = cls.execute(*args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/interfaces/video_upscaler.py", line 580, in execute
    raise e
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/interfaces/video_upscaler.py", line 497, in execute
    ctx = decode_all_batches(
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/core/generation_phases.py", line 937, in decode_all_batches
    samples = runner.vae_decode([upscaled_latent])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/core/infer.py", line 262, in vae_decode
    sample = self.vae.decode(
             ^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1688, in decode
    x = super().decode(z, return_dict=return_dict, tiled=tiled, tile_size=tile_size,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/diffusers/utils/accelerate_utils.py", line 46, in wrapper
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1203, in decode
    decoded = self.tiled_decode(z, tile_size=tile_size, tile_overlap=tile_overlap)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1557, in tiled_decode
    decoded_tile = self.slicing_decode(tile_latent)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1292, in slicing_decode
    return self._decode(z)
           ^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1240, in _decode
    output = self.decoder(_z, memory_state=memory_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 1028, in forward
    sample = up_block(sample, latent_embeds, memory_state=memory_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 553, in forward
    hidden_states = resnet(hidden_states, temb=None, memory_state=memory_state)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/attn_video_vae.py", line 336, in forward
    hidden_states = self.conv1(hidden_states, memory_state=memory_state)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 227, in forward
    return self.slicing_forward(input, memory_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 296, in slicing_forward
    input[i] = self.memory_limit_conv(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/models/video_vae_v3/modules/causal_inflation_lib.py", line 204, in memory_limit_conv
    output = retry_on_oom(
             ^^^^^^^^^^^^^
  File "/Users/peter/Documents/ComfyUI/custom_nodes/seedvr2_videoupscaler/src/optimization/memory_manager.py", line 394, in retry_on_oom
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
RuntimeError: MPS backend out of memory (MPS allocated: 18.69 GB, other allocations: 113.43 GB, max allowed: 132.19 GB). Tried to allocate 128.00 MB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).

2025-12-13T19:47:28.404443 - Prompt executed in 279.70 seconds

```
## Attached Workflow
Please make sure that workflow does not contain any sensitive information such as API keys or passwords.
```
{"id":"e2c8e6ec-9e8a-4a81-a240-7e7d2befc2bf","revision":0,"last_node_id":21,"last_link_id":19,"nodes":[{"id":19,"type":"SeedVR2TorchCompileSettings","pos":[-366.25065217175177,401.8303561599628],"size":[307.671484375,178],"flags":{},"order":0,"mode":4,"inputs":[{"localized_name":"backend","name":"backend","type":"COMBO","widget":{"name":"backend"},"link":null},{"localized_name":"mode","name":"mode","type":"COMBO","widget":{"name":"mode"},"link":null},{"localized_name":"fullgraph","name":"fullgraph","type":"BOOLEAN","widget":{"name":"fullgraph"},"link":null},{"localized_name":"dynamic","name":"dynamic","type":"BOOLEAN","widget":{"name":"dynamic"},"link":null},{"localized_name":"dynamo_cache_size_limit","name":"dynamo_cache_size_limit","type":"INT","widget":{"name":"dynamo_cache_size_limit"},"link":null},{"localized_name":"dynamo_recompile_limit","name":"dynamo_recompile_limit","type":"INT","widget":{"name":"dynamo_recompile_limit"},"link":null}],"outputs":[{"localized_name":"TORCH_COMPILE_ARGS","name":"TORCH_COMPILE_ARGS","type":"TORCH_COMPILE_ARGS","links":[18,19]}],"properties":{"Node name for S&R":"SeedVR2TorchCompileSettings","aux_id":"ainvfx/ComfyUI-SeedVR2_VideoUpscaler","ver":"690cc39379c1481159ddd451368dbf2295930fc6"},"widgets_values":["inductor","default",false,false,64,128]},{"id":18,"type":"Note","pos":[420.71483854947945,-81.0108307454865],"size":[210,94.05179298404067],"flags":{},"order":1,"mode":0,"inputs":[],"outputs":[],"properties":{},"widgets_values":["Enable to upscale alpha/mask channel along with RGB channel (for RGBA inputs)"],"color":"#432","bgcolor":"#653"},{"id":17,"type":"JoinImageWithAlpha","pos":[449.4979840435381,-180.32741340177097],"size":[176.86484375,46],"flags":{},"order":7,"mode":0,"inputs":[{"localized_name":"image","name":"image","type":"IMAGE","link":14},{"localized_name":"alpha","name":"alpha","type":"MASK","link":17}],"outputs":[{"localized_name":"IMAGE","name":"IMAGE","type":"IMAGE","links":[16]}],"properties":{"Node name for S&R":"JoinImageWithAlpha","cnr_id":"comfy-core","ver":"0.3.68"},"widgets_values":[]},{"id":15,"type":"SaveImage","pos":[1167.7234520647182,192.0920014495034],"size":[270,270],"flags":{},"order":9,"mode":0,"inputs":[{"localized_name":"images","name":"images","type":"IMAGE","link":12},{"localized_name":"filename_prefix","name":"filename_prefix","type":"STRING","widget":{"name":"filename_prefix"},"link":null}],"outputs":[],"properties":{"cnr_id":"comfy-core","ver":"0.3.68"},"widgets_values":["ComfyUI"]},{"id":20,"type":"Note","pos":[-353.44461861972593,641.0883773827939],"size":[290.1922351462714,97.95592567265356],"flags":{},"order":2,"mode":0,"inputs":[],"outputs":[],"properties":{},"widgets_values":["Enable torch.compile optimization. First batch has compilation overhead, but speeds up subsequent batches. Best for long videos, not single images."],"color":"#432","bgcolor":"#653"},{"id":16,"type":"LoadImage","pos":[80.5773583147188,-182.2697173004957],"size":[274.080078125,314],"flags":{},"order":3,"mode":0,"inputs":[{"localized_name":"image","name":"image","type":"COMBO","widget":{"name":"image"},"link":null},{"localized_name":"choose file to upload","name":"upload","type":"IMAGEUPLOAD","widget":{"name":"upload"},"link":null}],"outputs":[{"localized_name":"IMAGE","name":"IMAGE","type":"IMAGE","links":[14]},{"localized_name":"MASK","name":"MASK","type":"MASK","links":[17]}],"properties":{"Node name for S&R":"LoadImage","cnr_id":"comfy-core","ver":"0.3.68"},"widgets_values":["dataset_00002_512.png","image"]},{"id":14,"type":"SeedVR2LoadDiTModel","pos":[64.4234520647185,212.7072358245033],"size":[307.6646484375,202],"flags":{},"order":5,"mode":0,"inputs":[{"localized_name":"torch_compile_args","name":"torch_compile_args","shape":7,"type":"TORCH_COMPILE_ARGS","link":18},{"localized_name":"model","name":"model","type":"COMBO","widget":{"name":"model"},"link":null},{"localized_name":"device","name":"device","type":"COMBO","widget":{"name":"device"},"link":null},{"localized_name":"blocks_to_swap","name":"blocks_to_swap","shape":7,"type":"INT","widget":{"name":"blocks_to_swap"},"link":null},{"localized_name":"swap_io_components","name":"swap_io_components","shape":7,"type":"BOOLEAN","widget":{"name":"swap_io_components"},"link":null},{"localized_name":"offload_device","name":"offload_device","shape":7,"type":"COMBO","widget":{"name":"offload_device"},"link":null},{"localized_name":"cache_model","name":"cache_model","shape":7,"type":"BOOLEAN","widget":{"name":"cache_model"},"link":null},{"localized_name":"attention_mode","name":"attention_mode","shape":7,"type":"COMBO","widget":{"name":"attention_mode"},"link":null}],"outputs":[{"localized_name":"SEEDVR2_DIT","name":"SEEDVR2_DIT","type":"SEEDVR2_DIT","links":[11]}],"properties":{"Node name for S&R":"SeedVR2LoadDiTModel","aux_id":"ainvfx/ComfyUI-SeedVR2_VideoUpscaler","ver":"690cc39379c1481159ddd451368dbf2295930fc6"},"widgets_values":["seedvr2_ema_7b_sharp_fp16.safetensors","mps",36,false,"cpu",false,"sdpa"]},{"id":13,"type":"SeedVR2LoadVAEModel","pos":[60.33985831471847,499.8798920745057],"size":[312.866796875,298],"flags":{},"order":6,"mode":0,"inputs":[{"localized_name":"torch_compile_args","name":"torch_compile_args","shape":7,"type":"TORCH_COMPILE_ARGS","link":19},{"localized_name":"model","name":"model","type":"COMBO","widget":{"name":"model"},"link":null},{"localized_name":"device","name":"device","type":"COMBO","widget":{"name":"device"},"link":null},{"localized_name":"encode_tiled","name":"encode_tiled","shape":7,"type":"BOOLEAN","widget":{"name":"encode_tiled"},"link":null},{"localized_name":"encode_tile_size","name":"encode_tile_size","shape":7,"type":"INT","widget":{"name":"encode_tile_size"},"link":null},{"localized_name":"encode_tile_overlap","name":"encode_tile_overlap","shape":7,"type":"INT","widget":{"name":"encode_tile_overlap"},"link":null},{"localized_name":"decode_tiled","name":"decode_tiled","shape":7,"type":"BOOLEAN","widget":{"name":"decode_tiled"},"link":null},{"localized_name":"decode_tile_size","name":"decode_tile_size","shape":7,"type":"INT","widget":{"name":"decode_tile_size"},"link":null},{"localized_name":"decode_tile_overlap","name":"decode_tile_overlap","shape":7,"type":"INT","widget":{"name":"decode_tile_overlap"},"link":null},{"localized_name":"tile_debug","name":"tile_debug","shape":7,"type":"COMBO","widget":{"name":"tile_debug"},"link":null},{"localized_name":"offload_device","name":"offload_device","shape":7,"type":"COMBO","widget":{"name":"offload_device"},"link":null},{"localized_name":"cache_model","name":"cache_model","shape":7,"type":"BOOLEAN","widget":{"name":"cache_model"},"link":null}],"outputs":[{"localized_name":"SEEDVR2_VAE","name":"SEEDVR2_VAE","type":"SEEDVR2_VAE","links":[10]}],"properties":{"Node name for S&R":"SeedVR2LoadVAEModel","aux_id":"ainvfx/ComfyUI-SeedVR2_VideoUpscaler","ver":"690cc39379c1481159ddd451368dbf2295930fc6"},"widgets_values":["ema_vae_fp16.safetensors","mps",true,1024,128,true,1024,128,"false","cpu",false]},{"id":10,"type":"SeedVR2VideoUpscaler","pos":[843.8992333147189,191.3224701995034],"size":[270,386],"flags":{},"order":8,"mode":0,"inputs":[{"localized_name":"image","name":"image","type":"IMAGE","link":16},{"localized_name":"dit","name":"dit","type":"SEEDVR2_DIT","link":11},{"localized_name":"vae","name":"vae","type":"SEEDVR2_VAE","link":10},{"localized_name":"seed","name":"seed","type":"INT","widget":{"name":"seed"},"link":null},{"localized_name":"resolution","name":"resolution","type":"INT","widget":{"name":"resolution"},"link":null},{"localized_name":"max_resolution","name":"max_resolution","type":"INT","widget":{"name":"max_resolution"},"link":null},{"localized_name":"batch_size","name":"batch_size","type":"INT","widget":{"name":"batch_size"},"link":null},{"localized_name":"uniform_batch_size","name":"uniform_batch_size","type":"BOOLEAN","widget":{"name":"uniform_batch_size"},"link":null},{"localized_name":"color_correction","name":"color_correction","type":"COMBO","widget":{"name":"color_correction"},"link":null},{"localized_name":"temporal_overlap","name":"temporal_overlap","shape":7,"type":"INT","widget":{"name":"temporal_overlap"},"link":null},{"localized_name":"prepend_frames","name":"prepend_frames","shape":7,"type":"INT","widget":{"name":"prepend_frames"},"link":null},{"localized_name":"input_noise_scale","name":"input_noise_scale","shape":7,"type":"FLOAT","widget":{"name":"input_noise_scale"},"link":null},{"localized_name":"latent_noise_scale","name":"latent_noise_scale","shape":7,"type":"FLOAT","widget":{"name":"latent_noise_scale"},"link":null},{"localized_name":"offload_device","name":"offload_device","shape":7,"type":"COMBO","widget":{"name":"offload_device"},"link":null},{"localized_name":"enable_debug","name":"enable_debug","shape":7,"type":"BOOLEAN","widget":{"name":"enable_debug"},"link":null}],"outputs":[{"localized_name":"IMAGE","name":"IMAGE","type":"IMAGE","links":[12]}],"properties":{"Node name for S&R":"SeedVR2VideoUpscaler","aux_id":"ainvfx/ComfyUI-SeedVR2_VideoUpscaler","ver":"690cc39379c1481159ddd451368dbf2295930fc6"},"widgets_values":[42,"fixed",4096,4096,1,false,"lab",0,0,0,0,"cpu",false]},{"id":21,"type":"StringCompare","pos":[1011.4753214618321,217.86274334201659],"size":[400,200],"flags":{"collapsed":true},"order":4,"mode":0,"inputs":[{"localized_name":"string_a","name":"string_a","type":"STRING","widget":{"name":"string_a"},"link":null},{"localized_name":"string_b","name":"string_b","type":"STRING","widget":{"name":"string_b"},"link":null},{"localized_name":"mode","name":"mode","type":"COMBO","widget":{"name":"mode"},"link":null},{"localized_name":"case_sensitive","name":"case_sensitive","type":"BOOLEAN","widget":{"name":"case_sensitive"},"link":null}],"outputs":[{"localized_name":"BOOLEAN","name":"BOOLEAN","type":"BOOLEAN","links":null}],"properties":{"Node name for S&R":"StringCompare"},"widgets_values":["","","Starts With",true]}],"links":[[10,13,0,10,2,"SEEDVR2_VAE"],[11,14,0,10,1,"SEEDVR2_DIT"],[12,10,0,15,0,"IMAGE"],[14,16,0,17,0,"IMAGE"],[16,17,0,10,0,"IMAGE"],[17,16,1,17,1,"MASK"],[18,19,0,14,0,"TORCH_COMPILE_ARGS"],[19,19,0,13,0,"TORCH_COMPILE_ARGS"]],"groups":[],"config":{},"extra":{"ds":{"scale":1.3569525417026709,"offset":[-449.92288938940237,122.9745267806852]},"workflowRendererVersion":"LG"},"version":0.4}
```

## Additional Context
(Please add any additional context or steps to reproduce the error here)

--- a/src/models/video_vae_v3/modules/attn_video_vae.py
+++ b/src/models/video_vae_v3/modules/attn_video_vae.py
@@ -51,7 +51,7 @@ from .types import (
     _memory_device_t,
     _receptive_field_t,
 )
-from ....optimization.memory_manager import retry_on_oom
+from ....optimization.memory_manager import is_mps_available, retry_on_oom
 
 logger = get_logger(__name__)  # pylint: disable=invalid-name
 
@@ -1224,6 +1224,9 @@ class VideoAutoencoderKL(diffusers.AutoencoderKL):
         
         output = causal_conv_gather_outputs(output)
         
+        if is_mps_available():
+            torch.mps.empty_cache()
+
         # Only transfer back if needed
         return output if output.device == x.device else output.to(x.device)
 
@@ -1240,6 +1243,9 @@ class VideoAutoencoderKL(diffusers.AutoencoderKL):
         output = self.decoder(_z, memory_state=memory_state)
         output = causal_conv_gather_outputs(output)
         
+        if is_mps_available():
+            torch.mps.empty_cache()
+
         # Only transfer back if needed
         return output if output.device == z.device else output.to(z.device)
 

